frontend/src/lib/utils/error.ts

```diff
@@ -0,0 +1,13 @@
+export class InvalidRelatedObjectError extends Error {
```

Comment:
This doesn't feel generic and I think it's fine to leave this next to the implementation?